nodeup/pkg/model/kube_proxy.go

```diff
@@ -124,9 +124,39 @@ func (b *KubeProxyBuilder) buildPod() (*v1.Pod, error) {
 		}
 	}
 
-	cpuRequest, err := resource.ParseQuantity(c.CPURequest)
-	if err != nil {
-		return nil, fmt.Errorf("Error parsing CPURequest=%q", c.CPURequest)
+	resourceRequests := v1.ResourceList{}
+	resourceLimits := v1.ResourceList{}
+
+	if c.CPURequest != "" {
+		cpuRequest, err := resource.ParseQuantity(c.CPURequest)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing CPURequest=%q", c.CPURequest)
+		}
+		resourceRequests["cpu"] = cpuRequest
+	}
+
+	if c.CPULimit != "" {
```

Comment:
Like we discussed before, if cpulimit is set, but cpurequest isnt we should throw an error too, same for memory.

Reply:
I'll omit this change since the limit for both memory and cpu can be set without cpu and memory request being set and vice versa. Also, to avoid coupling the kops api validation logic to the k8s api.